### PR TITLE
WS-827: define the classes for the batch data model.

### PR DESCRIPTION
**Pull request by benson-basis**

### CODE REVIEW

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * used in {@link Request#contentType} identifies
+ * a json file consisting of an array of strings; each string being treated as an
+ * input. The string length in this case is limited to 200 characters.
+ */
+public class BatchRequestItem {
+    private final String endpoint;
+    //TODO: make this, well, 'Object', or an empty abstract class, and set up polymorphism.
+    private final Request documentRequest;
+    private final NameTranslationRequest nameTranslationRequest;
+    private final NameSimilarityRequest nameSimilarityRequest;
```
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java:36]: I'd like to do the TODO with an empty base class. This is confusing as is!

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Define a batch job. Serialize this object to json to create a file
+ * that requests a job.
+ */
+public class BatchRequest {
+    // user-specified ID for the overall batch.
+    private final String batchId;
+    // URL to post when the entire job is complete.
+    private final String completionCallbackUrl;
+    // Define the items that make up the batch.
+    private final BatchRequestItem[] items;
```
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java:28]: I always prefer List over array, anyone else?

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Define a batch job. Serialize this object to json to create a file
+ * that requests a job.
+ */
+public class BatchRequest {
+    // user-specified ID for the overall batch.
+    private final String batchId;
+    // URL to post when the entire job is complete.
+    private final String completionCallbackUrl;
```
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java:26]: If we call it a `XyzUrl`, should it be a `URL` type?

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * One item in a batch request. Each item consists of an input
+ * to process and an endpoint to apply to the item.
+ * The inputs are specified by {@link com.basistech.rosette.apimodel.Request} objects.
+ * The special content type {@code application/vnd.basistech-multiple-inputs}
+ * used in {@link Request#contentType} identifies
+ * a json file consisting of an array of strings; each string being treated as an
+ * input. The string length in this case is limited to 200 characters.
+ */
+public class BatchRequestItem {
```
Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java:31]: We don't have an ID field in the Request class, we either need to add to that or here.

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java (hunk 1 above)
Comment (benson-basis) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java:36]: It's a big change to the existing request classes, for which there is a JIRA.

Comment (guruxu) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequestItem.java:36]: RCB-401, okay.

File: model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java (hunk 2 above)
Comment (jkho) [inline model/src/main/java/com/basistech/rosette/apimodel/batch/BatchRequest.java:28]: What if customer submit 2 BatchRequests with the same batchId? I think we need to make up an internally unique batchId.

**Comment by guruxu** [comment]
Closing and use @jkho 's PR

**Pull request closed without merging**